deltachat-ios/Controller/WebxdcViewController.swift

```diff
@@ -591,3 +587,45 @@ extension WebxdcViewController: WKURLSchemeHandler {
     func webView(_ webView: WKWebView, stop urlSchemeTask: WKURLSchemeTask) {
     }
 }
+
+// MARK: Memory Leak Prevention
+
+extension WKUserContentController {
+    func add(weak scriptMessageHandler: any WKScriptMessageHandler, name: String) {
+        add(WeakScriptMessageHandler(scriptMessageHandler), name: name)
+    }
+
+    private class WeakScriptMessageHandler: NSObject, WKScriptMessageHandler {
```

Comment:
cool, this `WKScriptMessageHandler` is a protocol - i was first wondering how we can know if we've overwritten/implemented all methods required.

Reply:
I'm still always paranoid about that because protocols can have optional requirements